src/components/dialog/dialog.spec.js

```diff
@@ -781,6 +781,34 @@ describe('$mdDialog', function() {
 
       expect(response).toBe('responsetext');
     }));
+
+    it('should not submit after ENTER key when input is empty and prompt is required', inject(function($mdDialog, $rootScope, $timeout, $mdConstant) {
```

Comment:
This is good, but I think that we should also have a test that: when required is set and the input has a value that hide is called and the result is correct.